ProductControllerTests.cs

```diff
@@ -0,0 +1,77 @@
+using CoreServices.Controllers;
+using CoreServices.DTO;
+using CoreServices.Repository;
+using CoreServices.Services;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace CoreServices.Tests
+{
+    public class ProductControllerTests
+    {
+        private readonly ProductController _productController;
+        private readonly ILogger _logger;
+        private readonly Mock<IRepository> _repository;
+        private readonly Mock<ProductService> _mockProductService;
+
+        public ProductControllerTests()
+        {
+            _repository = new Mock<IRepository>();
+            _mockProductService = new Mock<ProductService>(_repository.Object);
+            _productController = new ProductController(_mockProductService.Object, _logger);
+        }
+
+
+        [Fact]
+        public async void Task_AddAsync_PassedValidData_Return_OkResult()
+        {
+            //Arrange
+            var product = new ProductDTO() { Id = 5, Name = "Test Name5", Price = 8120 };
+            _mockProductService.Setup(p => p.AddAsync(product)).ReturnsAsync(product);
+
+            //Act  
+            var data = await _productController.AddAsync(product);
+
+            //Assert
+            var resulttype = Assert.IsType<CreatedAtActionResult>(data);
+            Assert.Equal(typeof(CreatedAtActionResult), resulttype.GetType());

```

Comment:
To validate result type:
Assert.IsType<CreatedAtActionResult>(data);
Please remove
 Assert.Equal(typeof(CreatedAtActionResult), resulttype.GetType());